sdk/src/main/java/com/kin/ecosystem/transfer/presenter/AccountInfoPresenter.java

```diff
@@ -0,0 +1,147 @@
+package com.kin.ecosystem.transfer.presenter;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import com.kin.ecosystem.base.BasePresenter;
+import com.kin.ecosystem.common.exception.BlockchainException;
+import com.kin.ecosystem.core.data.blockchain.BlockchainSourceImpl;
+import com.kin.ecosystem.transfer.AccountInfoManager;
+import com.kin.ecosystem.transfer.view.IAccountInfoView;
+
+public class AccountInfoPresenter extends BasePresenter<IAccountInfoView> implements IAccountInfoPresenter {
+    private static final String EXTRA_SOURCE_APP_NAME = "EXTRA_SOURCE_APP_NAME";
+
+    private AccountInfoManager accountInfoManager;
+    private AccountInfoAsyncTask asyncTask;
+    private boolean taskComplete = false;
+    private boolean completeWithSuceess = false;
+    private boolean isPaused = false;
+
+    public AccountInfoPresenter(@NonNull AccountInfoManager accountInfoManager, @NonNull IAccountInfoView view, @NonNull Intent intent) {
+        this.accountInfoManager = accountInfoManager;
+        processIntent(view, intent);
+        view.attachPresenter(this);
+        startInit();
+    }
+
+    private void processIntent(IAccountInfoView view, Intent intent) {
+        if (intent != null && intent.hasExtra(EXTRA_SOURCE_APP_NAME)) {
+            String sourceApp = intent.getStringExtra(EXTRA_SOURCE_APP_NAME);
+            if (!sourceApp.isEmpty()) {
```

Comment:
check sourceApp != null

Reply:
i check that you cant pass null in the calling activity
hence it could be only empty